### PR TITLE
OOD-11: admin can fetch users from Sisu and add to Oodikone user-db

### PR DESCRIPTION
**Pull request by rundtjan**
https://funidata.atlassian.net/browse/OOD-11

### CODE REVIEW

File: services/frontend/src/components/Users/NewUserSection.jsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+      {showAdded && addedUser && <h4> Added user to the Oodikone user-database.</h4>}
+
+      {!isErrorGetUser && user && (
+        <SortableTable
+          columns={[
+            {
+              key: 'NAME',
+              title: 'Name',
+              getRowVal: user => {
```
Comment (mattirasanen) [inline services/frontend/src/components/Users/NewUserSection.jsx:75]: I think we could add here `sortable: false` and `filterable: false` to each column object

File: services/frontend/src/components/Users/NewUserSection.jsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+              getRowVal: () => (
+                <Button color="red" onClick={cancelUser}>
+                  Cancel
+                </Button>
+              ),
+            },
+          ]}
+          data={[user]}
+          hideHeaderBar
+          singleLine={false}
```
Comment (mattirasanen) [inline services/frontend/src/components/Users/NewUserSection.jsx:111]: Also we could check what does the `stech` props do

**Review by mattirasanen — APPROVED** [review]
Looking fetch users from sisu to me. But lets wait atleast one more fd review

File: services/frontend/src/components/Users/NewUserSection.jsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      setShowAdded(true)
+      setTimeout(() => setShowAdded(false), 2000)
+    }
+  }, [addedUser, isLoadingAddUser, isErrorAddUser])
+
+  const handleEppnOnChange = (_, { value }) => {
+    if (value) setEppn(value)
+  }
+
+  const getUser = event => {
```
Comment (VoxBorealis) [inline services/frontend/src/components/Users/NewUserSection.jsx:38]: Is it possible to call `getUser()` while `eppn` state = ´´? If so, is there a need to validate it before calling `getUserFromSisuByEppnQuery(eppn)`?

Most likely a non-issue

File: services/frontend/src/components/Users/NewUserSection.jsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    setUser('')
+    event.preventDefault()
+    getUserFromSisuByEppnQuery(eppn)
+  }
+
+  const addUser = () => {
+    addUserMutation(user)
```
Comment (VoxBorealis) [inline services/frontend/src/components/Users/NewUserSection.jsx:45]: Same thing here with the `user` state

**Review by VoxBorealis — APPROVED** [review]
LGTM! 👍

File: services/frontend/src/components/Users/NewUserSection.jsx (hunk 3 above)
Comment (rundtjan) [reply services/frontend/src/components/Users/NewUserSection.jsx:34]: Added an if (!eppn) return, just to be sure.

File: services/frontend/src/components/Users/NewUserSection.jsx (hunk 4 above)
Comment (rundtjan) [reply services/frontend/src/components/Users/NewUserSection.jsx:43]: Added an if (!user) return.

File: services/backend/src/routes/users.js [hunk 5]
```diff
@@ -83,4 +84,19 @@ router.post('/language', async (req, res) => {
...
+router.get('/from-sisu-by-eppn/:newUserEppn', auth.roles(['admin']), async (req, res) => {
+  let { username: requesterEppn } = req.user
+  const { newUserEppn } = req.params
+  // in order to test this feature in the dev environment we need to set an eppn that demo sisu will recognize
+  if (isDev && requesterEppn === 'mluukkai') requesterEppn = newUserEppn
+  const person = await userService.getUserFromSisuByEppn(requesterEppn, newUserEppn)
+  res.json(person)
+})
+
+router.post('/add', async (req, res) => {
```
Comment (rikurauhala) [inline services/backend/src/routes/users.js:96]: Should this route also contain `auth.roles(['admin'])`?

File: services/backend/src/routes/users.js
Comment (rikurauhala) [inline services/backend/src/routes/users.js:1]: We do not want/need these routes at Toska. Please add them only for service provider Funidata.

**Review by rikurauhala — CHANGES_REQUESTED** [review]
See comments.

File: services/backend/src/routes/users.js
Comment (mattirasanen) [reply services/backend/src/routes/users.js:1]: Should be fine now

File: services/backend/src/routes/users.js (hunk 5 above)
Comment (rundtjan) [reply services/backend/src/routes/users.js]: Yup, fixed that!

File: services/backend/src/routes/users.js
Comment (rikurauhala) [reply services/backend/src/routes/users.js:1]: Thank you!

**Review by rikurauhala — APPROVED** [review]

**Pull request merged**